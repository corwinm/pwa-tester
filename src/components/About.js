import React from "react";
import Page from "components/Page";

export default function About() {
  return (
    <Page title="About PWA Tester">
      <h2 className="subtitle">
        PWA Tester is a Reactjs Progressive Web App (PWA) single page application (spa) bootstrapped with
        Create React App and hosted with GithubPages.
      </h2>
      <p>This app was created with the following goals:</p>
      <ol style={{ listStylePosition: "inside" }}>
        <li>Exercise the basic features of a Progressive Web App (PWA) with React's hooks api.</li>
        <li>Enable testing PWA features across devices.</li>
        <li>Establish best practices for PWA apps that are frequently updated.</li>
        <li>Establish best practices for PWA apps analytics.</li>
        <li>Serve as a reference for myself or others to use when adding PWA features to an app.</li>
      </ol>
    </Page>
  );
}
