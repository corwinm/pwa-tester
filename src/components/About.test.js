import React from "react";
import { render, cleanup } from "react-testing-library";
import About from "./About";

afterEach(cleanup);

test("matches about snapshot", () => {
  const { container } = render(<About />);
  expect(container.firstChild).toMatchSnapshot();
});
