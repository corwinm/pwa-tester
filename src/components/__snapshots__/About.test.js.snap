// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches about snapshot 1`] = `
<div
  class="notch-inset-item"
>
  <section
    class="section"
  >
    <div
      class="container is-fluid"
    >
      <h1
        class="title has-text-white"
      >
        About PWA Tester
      </h1>
      <div>
        <h2
          class="subtitle"
        >
          PWA Tester is a Reactjs Progressive Web App (PWA) single page application (spa) bootstrapped with Create React App and hosted with GithubPages.
        </h2>
        <p>
          This app was created with the following goals:
        </p>
        <ol
          style="list-style-position: inside;"
        >
          <li>
            Exercise the basic features of a Progressive Web App (PWA) with React's hooks api.
          </li>
          <li>
            Enable testing PWA features across devices.
          </li>
          <li>
            Establish best practices for PWA apps that are frequently updated.
          </li>
          <li>
            Establish best practices for PWA apps analytics.
          </li>
          <li>
            Serve as a reference for myself or others to use when adding PWA features to an app.
          </li>
        </ol>
      </div>
    </div>
  </section>
</div>
`;
